### PR TITLE
Error handling

### LINKED ISSUE #7: Error handling in APIs
I noticed when setting up the repository that I was getting errors because some endpoints couldn't parse the response they got as json. The reason was that I didn't have client_id and client_secret in the env.

I think we can improve the error handling in some of our APIs.

### PR DESCRIPTION
**Pull request by dmahely**
I added a component called `<Container />` that renders `<App />` within `<ErrorBoundary />`, which in turn catches errors and renders an `<Error />` component in case any are found.

Next steps: 
1. Design nice UI for `<Error />` and implement it
2. Figure out what the 'Try again' button in it should do, I assume just refresh the page?

### CODE REVIEW

**Comment by tryggvigy** [comment]
> 2. Figure out what the 'Try again' button in it should do, I assume just refresh the page?

Yeah I think that makes sense!

File: src/components/Error/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { Error } from './Error';
+
+export { Error };
```
Comment (tryggvigy) [inline src/components/Error/index.js:3]: You should be able to do `export { Error } from './Error';` if you want

Comment (dmahely) [reply src/components/Error/index.js:3]: Neat! I got this trick from a post somewhere and didn't think much about improving it. I'll change the index files soon.

**Comment by dmahely** [comment]
I spent too much time trying to draw something to put in the error page, but ended up going another way! I added the design in Figma and implemented it here. I've been using `throw new Error('whatever')` in App's useEffect method to see the error page. There's a bit of white space (24px exactly) at the top in that page only that I can't figure out where it's coming from.

File: src/components/Error/Error.js [hunk 2]
```diff
@@ -9,7 +11,15 @@ const Error = (hasError) => {
     };
     return (
         <div className="Error--container">
-            <h1 className="Error--text">Something went wrong.</h1>
+            <h1 className="Error--heading">Oh no!</h1>
+            <img
+                className="Error--image"
+                src={albumCover}
+                alt="Please don't stop the music song cover"
```
Comment (tryggvigy) [inline src/components/Error/Error.js:18]: nice alt tag! :) those are important when it's possible to define them properly

**Review by tryggvigy — APPROVED** [review]

**Comment by dmahely** [comment]
Finally figured out where the pesky extra margin is coming from!

**Pull request merged**